File: hw/bonilla/U3/HW23Strategy/Strategy/src/main/java/ec/edu/espe/strategy/controller/BubbleSort.java
package ec.edu.espe.strategy.controller;

/**
 *
 * @author Bonilla David SoftCrafters
 */
public class BubbleSort implements SortingStrategy {

    @Override
    public int[] sort(int data[]) {
        System.out.println("Usando Bubble Sort...");
        int n = data.length;
        int[] sortedData = data.clone();

        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (sortedData[j] > sortedData[j + 1]) {
                    int temp = sortedData[j];
                    sortedData[j] = sortedData[j + 1];
                    sortedData[j + 1] = temp;
                }
            }
        }

        mostrarArreglo(sortedData);
        return sortedData;
    }

    private void mostrarArreglo(int[] data) {
        System.out.print("Resultado: ");
        for (int num : data) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}

File: hw/bonilla/U3/HW23Strategy/Strategyjs/controller/BubbleSort.js
const SortingStrategy = require('./SortingStrategy');

class BubbleSort extends SortingStrategy {
    sort(data) {
        console.log("Usando Bubble Sort...");
        const sortedData = [...data];
        const n = sortedData.length;

        for (let i = 0; i < n - 1; i++) {
            for (let j = 0; j < n - i - 1; j++) {
                if (sortedData[j] > sortedData[j + 1]) {
                    const temp = sortedData[j];
                    sortedData[j] = sortedData[j + 1];
                    sortedData[j + 1] = temp;
                }
            }
        }

        this.mostrarArreglo(sortedData);
        return sortedData;
    }

    mostrarArreglo(data) {
        process.stdout.write("Resultado: ");
        data.forEach(num => process.stdout.write(num + " "));
        console.log();
    }
}

module.exports = BubbleSort;
